middle.hpp

```diff
@@ -17,15 +17,17 @@ struct options_t;
 struct middle_query_t {
     virtual ~middle_query_t() {}
 
-    virtual int nodes_get_list(nodelist_t &out, const idlist_t nds) const = 0;
+    virtual size_t nodes_get_list(nodelist_t &out, const idlist_t nds) const = 0;
 
-    virtual int ways_get(osmid_t id, taglist_t &tags, nodelist_t &nodes) const = 0;
+    // returns true if it got the relation
```

Comment:
What's the best way to have API documentation for functions like this? This was a note from the cleanup that I decided to leave in.